### PR TITLE
Add equatorial compression option for spherical torus map 

### PR DESCRIPTION
**Pull request by jyoo1042**
## Proposed changes

Add an equatorial compression option for the spherical torus map using a simple cubic function. This is for PolarMagFmDisk runs which use spherical torus.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.cpp [hunk 1]
```diff
@@ -58,8 +71,9 @@ tnsr::I<T, 3> SphericalTorus::operator()(
     const tnsr::I<T, 3>& source_coords) const {
   tnsr::I<T, 3> result;
   const auto r = radius(get<0>(source_coords));
-  const auto theta =
-      M_PI_2 - (pi_over_2_minus_theta_min_ * get<1>(source_coords));
+  // intermediate variable for equatorial compression
+  const auto intermediate_val = cubic_compression(get<1>(source_coords));
```
Comment (wthrowe) [inline src/PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.cpp]: Having a variable called `intermediate_val` that's only used once isn't adding clarity.  Just call the function in the expression on the next line.

File: src/PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.cpp [hunk 2]
```diff
@@ -94,8 +116,10 @@ Jacobian<T, 3, Frame::BlockLogical, Frame::Inertial> SphericalTorus::jacobian(
 
   auto& sin_theta = get<2, 1>(jacobian);
   auto& cos_theta = get<2, 0>(jacobian);
+  // intermediate variable for equatorial compression
+  get<2, 2>(jacobian) = cubic_compression(get<1>(source_coords));
```
Comment (wthrowe) [inline src/PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.cpp]: Again, just call this inline.

File: src/PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.cpp [hunk 3]
```diff
@@ -135,8 +167,10 @@ SphericalTorus::inv_jacobian(const tnsr::I<T, 3>& source_coords) const {
 
   auto& sin_theta = get<1, 2>(inv_jacobian);
   auto& cos_theta = get<0, 2>(inv_jacobian);
+  // intermediate variable for equatorial compression
+  get<2, 2>(inv_jacobian) = cubic_compression(get<1>(source_coords));
```
Comment (wthrowe) [inline src/PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.cpp]: Again.  Go through and find all cases of this.

File: src/PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.cpp [hunk 4]
```diff
@@ -335,6 +415,24 @@ bool operator==(const SphericalTorus& lhs, const SphericalTorus& rhs) {
...
+template <typename T>
+T SphericalTorus::cubic_compression(const T& x) const {
+  return compression_level_ * pow<3>(x) + (1.0 - compression_level_) * x;
+}
+
+double SphericalTorus::cubic_inversion(const double& x) const {
+  // using p.228 of Numerical Recipe (cubic equation)
+  const double Q = (compression_level_ - 1.0) / (3.0 * compression_level_);
+  const double R = -0.5 * x / (compression_level_);
+  const double A =
```
Comment (wthrowe) [inline src/PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.cpp]: Lowercase variable names.

File: src/PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.cpp [hunk 5]
```diff
@@ -335,6 +415,24 @@ bool operator==(const SphericalTorus& lhs, const SphericalTorus& rhs) {
          lhs.fraction_of_torus_ == rhs.fraction_of_torus_;
 }
 
+template <typename T>
+T SphericalTorus::cubic_compression(const T& x) const {
+  return compression_level_ * pow<3>(x) + (1.0 - compression_level_) * x;
+}
+
+double SphericalTorus::cubic_inversion(const double& x) const {
```
Comment (wthrowe) [inline src/PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.cpp]: Take double by value, not reference.

File: tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_SphericalTorus.cpp [hunk 6]
```diff
@@ -157,6 +187,7 @@ SPECTRE_TEST_CASE("Unit.PointwiseFunctions.AnalyticData.GrMhd.SphericalTorus",
           const auto numerical_hessian = numerical_derivative(
               [&i, &j, &partial_torus](const std::array<double, 3>& y) {
                 const auto y_tnsr = a2t(y);
+                CAPTURE(y);
```
Comment (wthrowe) [inline tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_SphericalTorus.cpp]: This doesn't do anything.  There are no checks before it goes out of scope.

**Review by wthrowe — CHANGES_REQUESTED** [review]

File: src/PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.cpp (hunk 1 above)
Comment (jyoo1042) [reply src/PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.cpp]: okay.

File: src/PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.cpp (hunk 2 above)
Comment (jyoo1042) [reply src/PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.cpp]: done.

File: src/PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.cpp (hunk 3 above)
Comment (jyoo1042) [reply src/PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.cpp]: done.

File: src/PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.cpp (hunk 4 above)
Comment (jyoo1042) [reply src/PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.cpp]: done.

File: src/PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.cpp (hunk 5 above)
Comment (jyoo1042) [reply src/PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.cpp]: done.

File: tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_SphericalTorus.cpp (hunk 6 above)
Comment (jyoo1042) [reply tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_SphericalTorus.cpp]: done.

File: src/PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.cpp [hunk 7]
```diff
@@ -71,8 +71,8 @@ tnsr::I<T, 3> SphericalTorus::operator()(
     const tnsr::I<T, 3>& source_coords) const {
   tnsr::I<T, 3> result;
   const auto r = radius(get<0>(source_coords));
-  const auto theta = M_PI_2 - (pi_over_2_minus_theta_min_ *
-                               cubic_compression(get<1>(source_coords)));
+  const T compressed_coord = cubic_compression(get<1>(source_coords));
+  const auto theta = M_PI_2 - (pi_over_2_minus_theta_min_ * compressed_coord);
   const auto phi = M_PI * fraction_of_torus_ * get<2>(source_coords);
```
Comment (wthrowe) [inline src/PointwiseFunctions/AnalyticData/GrMhd/SphericalTorus.cpp]: Change these `auto`s to `T`s, and then it should work.

**Review by wthrowe — CHANGES_REQUESTED** [review]

**Comment by wthrowe** [comment]
You have a syntax error typo, but otherwise looks good.  You can squash when you fix it.

**Review by wthrowe — APPROVED** [review]

**Pull request merged**